### PR TITLE
Cleaning up formatting for nested TypeArguments

### LINKED ISSUE #295: Nested Generics in Variable Declaration
I'm not exactly sure what to do with this, but it doesn't look the best right now.
```
class ClassName
{
    void MethodName()
    {
        PropertyMapActions.Add(
            pm =>
            {
                Expression<Func<
                        TSource,
                        TDestination,
                        TMember,
                        TMember,
                        ResolutionContext,
                        bool
                    >> expr = (src, dest, srcMember, destMember, ctxt) => condition(
                    src,
                    dest,
                    srcMember,
                    destMember,
                    ctxt
                );
            }
        );
    }
}

### PR DESCRIPTION
**Pull request by belav**
closes #295

### CODE REVIEW

File: Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/TypeArgumentList.cs [hunk 1]
```diff
@@ -9,13 +9,16 @@ public static class TypeArgumentList
     {
         public static Doc Print(TypeArgumentListSyntax node)
         {
+            Doc separator = node.Arguments.Count > 1
+            || node.Arguments.Any(o => o is GenericNameSyntax) ? Doc.SoftLine : Doc.Null;
```
Comment (shocklateboy92) [inline Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/TypeArgumentList.cs:13]: Odd. Why didn't this indent?

File: Src/CSharpier.Tests/TestFiles/ObnoxiousEdgeCases/ObnoxiousEdgeCases.cst [hunk 2]
```diff
@@ -2,15 +2,13 @@
 // try to clean these up if you are modifying formatting that affects them.
```
Comment (shocklateboy92) [inline Src/CSharpier.Tests/TestFiles/ObnoxiousEdgeCases/ObnoxiousEdgeCases.cst:2]: Is this comment still necessary now that you've cleaned these up, or do you have more cleanup you intend to do?

File: Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/TypeArgumentList.cs (hunk 1 above)
Comment (belav) [reply Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/TypeArgumentList.cs:13]: I just checked this in 0.9.8 and it looks better now.
There is also #345 for improving some of the edge cases.

File: Src/CSharpier.Tests/TestFiles/ObnoxiousEdgeCases/ObnoxiousEdgeCases.cst (hunk 2 above)
Comment (belav) [reply Src/CSharpier.Tests/TestFiles/ObnoxiousEdgeCases/ObnoxiousEdgeCases.cst:2]: Just glancing at them, there are probably still some improvements to be made. I mostly have the comment here so that people don't think this is really how we want things to look. If I ever get around to #272 this file can go away completely and they'll turn into proper test cases
